cgi-bin/LJ/Web.pm

```diff
@@ -2738,19 +2738,47 @@ sub control_strip
                     'trusted_by'        => BML::ml('web.controlstrip.status.trustedby', {'user' => $journal_display}),
                     'watched_by'        => BML::ml('web.controlstrip.status.watchedby', {'user' => $journal_display}),
                     );
-    # Style the status text
-    foreach my $key (keys %statustext) {
-        $statustext{$key} = "<span id='lj_controlstrip_statustext'>" . $statustext{$key} . "</span>";
-    }
 
-    my $ret;
+    # Vars for controlstrip.tt
+    my $template_args = {
+        'view' => $view,
+        'userpic_html' => '',
+        'userpic_class' => 'lj_controlstrip_userpic',
+        'logo_html' => ( LJ::Hooks::run_hook( 'control_strip_logo', $remote, $journal ) || '' ),
+        'show_login_form' => 0,
+        'login_chal' => '',
+        'links' => \%links,
+        'statustext' => '',
+        # remote # only set if logged in
+            # .user
+            # .sessid
+            # .display
+            # .is_validated
+            # .is_identity
+        'actionlinks' => [],
+        # filters # only set if viewing reading or network page
+            # .all => []
+            # .selected => ""
+        'viewoptions' => [],
+        'search_html' => LJ::Widget::Search->render,
+    };
+
+    # Shortcuts for the two nested array refs that get repeatedly dereferenced later
+    my $actionlinks = $template_args->{'actionlinks'};
+    my $viewoptions = $template_args->{'viewoptions'};
+
     if ($remote) {
-        my $remote_display  = LJ::ljuser($remote);
         my $userpic = $remote->userpic;
+        $template_args->{'remote'} = {
+            'sessid' => $remote->{'_session'}->{'sessid'} || 0,
+            'user' => $remote->{'user'},
+            'display' => LJ::ljuser($remote),
```

Comment:
The `LJ::ljuser` function is deprecated in favor of the equivalent user method `$remote->ljuser_display`, so we should update that here.

We also discourage treating the user object as a hash, so `$remote->{'user'}` and `$remote->{'_session'}->{'sessid'}` should be updated to `$remote->user` and `$remote->session->{'sessid'}` going forward.  (Or even `$remote->session->id` if you want to make sure the session is defined first -- attempting to call methods on undefined objects is the most frequent problem revealed when doing this conversion.)

Reply:
Oh!! Haha, I totally thought it WAS a hash. Or, I guess I had to switch the `is_verified`/`is_identity` things to method syntax before they'd work, so probably I thought it was something like a javascript object where it has both properties and methods. 

What's the difference between `$remote->session->{'sessid'}` and `$remote->session->id`?

Reply:
Basically equivalent, just expressed as a hash value versus using an object method.  Objects in Perl are weird, they’re fundamentally “blessed” hashes that you can call methods on.  But accessing a value of an undefined hash complains quietly, while calling a method on an undefined object throws a tantrum that will stop the execution flow.